### PR TITLE
IMPROVEMENT: wrapped values to member expressions

### PR DESCRIPTION
**Pull request by KyMback**
- added wrapping to member expressions
- bumped System.Text.Json, because "6.0.5" has high severity vulnerability

### CODE REVIEW

File: src/Webinex.Asky/Factories/FilterExpressions.cs [hunk 1]
```diff
@@ -198,4 +215,41 @@ public static Expression<Func<TEntity, bool>> All<TEntity>(
...
+    /// It make sense to wrap value to container instead of using Expression.Constant directly, because in this case,
+    /// tools like Entity Framework later will be able to optimize it.
+    /// </summary>
+    private static MemberExpression WrapValueToContainerMember(object value, Type type)
+    {
+        var accessor = ValueContainerFactoriesCache.GetOrAdd(type, (Type t) =>
+        {
+            var param = Expression.Parameter(typeof(object));
+            var ctorInfo = typeof(ValueContainer<>).MakeGenericType(t)
+                .GetConstructor(new[] { t }) ?? throw new ArgumentNullException();
```
Comment (s-skalaban-itra) [inline src/Webinex.Asky/Factories/FilterExpressions.cs:229]: it's InvalidOperationException or may be anything else, but definitely not a ArgumentNullException

File: src/Webinex.Asky/Factories/FilterExpressions.cs [hunk 2]
```diff
@@ -198,4 +215,41 @@ public static Expression<Func<TEntity, bool>> All<TEntity>(
...
+        var containerInstance = accessor(value);
+        return Expression.MakeMemberAccess(
+            Expression.Constant(containerInstance),
+            containerInstance.GetType().GetMember(nameof(ValueContainer<object>.Value)).First());
+    }
+
+    /// <summary>
+    /// It make no sense to use struct here, because it will be anyway allocated in the heap
+    /// </summary>
+    private sealed class ValueContainer<TValue>
```
Comment (s-skalaban-itra) [inline src/Webinex.Asky/Factories/FilterExpressions.cs:246]: btw, you can use Tuple<X>

File: src/Webinex.Asky/Factories/FilterExpressions.cs [hunk 3]
```diff
@@ -198,4 +215,41 @@ public static Expression<Func<TEntity, bool>> All<TEntity>(
...
+            var expression = Expression.Lambda<Func<object, object>>(
+                Expression.New(ctorInfo, new[] { Expression.Convert(param, t) }),
+                new[] { param });
+            return expression.Compile();
+        });
+
+        var containerInstance = accessor(value);
+        return Expression.MakeMemberAccess(
+            Expression.Constant(containerInstance),
+            containerInstance.GetType().GetMember(nameof(ValueContainer<object>.Value)).First());
```
Comment (s-skalaban-itra) [inline src/Webinex.Asky/Factories/FilterExpressions.cs:240]: containerInstance.GetType().GetMember(nameof(ValueContainer<object>.Value)) can be cached as well (if needed)

File: src/Webinex.Asky/Factories/FilterExpressions.cs [hunk 4]
```diff
@@ -198,4 +215,41 @@ public static Expression<Func<TEntity, bool>> All<TEntity>(
...
+    private static MemberExpression WrapValueToContainerMember(object value, Type type)
+    {
+        var accessor = ValueContainerFactoriesCache.GetOrAdd(type, (Type t) =>
+        {
+            var param = Expression.Parameter(typeof(object));
+            var ctorInfo = typeof(ValueContainer<>).MakeGenericType(t)
+                .GetConstructor(new[] { t }) ?? throw new ArgumentNullException();
+
+            var expression = Expression.Lambda<Func<object, object>>(
+                Expression.New(ctorInfo, new[] { Expression.Convert(param, t) }),
```
Comment (s-skalaban-itra) [inline src/Webinex.Asky/Factories/FilterExpressions.cs:232]: is it possible to avoid Convert?

**Review by s-skalaban-itra — APPROVED** [review]

File: src/Webinex.Asky/Factories/FilterExpressions.cs (hunk 4 above)
Comment (KyMback) [reply src/Webinex.Asky/Factories/FilterExpressions.cs:232]: we need to have correct type in container, so we need this convert anyway

**Review by s-skalaban-itra — APPROVED** [review]

**Review by webinex — APPROVED** [review]

**Pull request merged**